code-challenges.test.js

```diff
@@ -14,48 +14,134 @@
 // Reminder: The test will call your function
 
 // --------------------1) Create a function that takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
-
+/////////////////////////////////////////////////// Todd's Pseudo Code ///////////////////////////////////////////////////
+// first step is to write a test and get a successful Fail
+// I know that my parameters are the secretCodeWords 1, 2 and 3
+// In the function that I wrote are used a couple of things The reduce function allows it to go through your strings and read it the accumulator accumulates it and as it accumulated the reducer sees that 
+// the current value of a = 4 and so forth and so forth accumulating those values and tell them you get the expected output.
+// This one threw me for a loop for at least a couple hours until I did my research and remembered the reduce function.
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 // a) Create a test with an expect statement using the variables provided.
-
-var secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
-var secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
-var secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
-
-
-
+describe("sercretCode", () => {
+    test("takes in a string and returns a coded message.", () => {
+        var secretCodeWord1 = "Lackadaisical"
+        // Expected output: "L4ck4d41s1c4l"
+        expect(sercretCode(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+        var secretCodeWord2 = "Gobbledygook"
+        // Expected output: "G0bbl3dyg00k" 
+        expect(sercretCode(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+        var secretCodeWord3 = "Eccentric"
+        // Expected output: "3cc3ntr1c"
+        expect(sercretCode(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+})
 // b) Create the function that makes the test pass.
+const sercretCode = (string) => {
 
+    const secretArr = string.split('');
+    const codedMessage = secretArr.reduce((accum, currVal) => {
```

Comment:
oooo very spiffy! using .reduce() instead of .map() and .join(). I think the power level on this method is quiet high for the problem at hand but congrats on figuring it out!